binderhub/app.py

```diff
@@ -196,6 +196,19 @@ def _valid_badge_base_url(self, proposal):
         config=True,
     )
 
+    sticky_builds = Bool(
+        False,
+        help="""
+        Attempt to assign builds for the same repository to the same node.
+
+        In order to speed up re-builds of a repository all its builds will
+        be assigned to the same node in the cluster.
+
+        Note: This feature only works if you also enable docker-in-docker support.
```

Comment:
Right now because we use the list of DIND pods to find the list of nodes. To support it without DIND pods we'd need a way to evaluate the node selector expression, which needs more access rights (listing all nodes).

Where would be a good place to document this thinking? The docs for the end user don't seem like the right place (I think it would be too much detail). Maybe in the code that implements the feature?

Reply:
I don't mind seeing it documented here, but if not, we could open an issue to represent this shortcoming could be overcome.